### ISSUE TITLE
[ci] base+async tends to fail often

### ISSUE THREAD

**Issue by ejgallego** [rocq-prover/rocq#9658]
As pointed out on gitter it is not rare to see this job randomly failing. Example:

- https://gitlab.com/coq/coq/-/jobs/167989714
  > `Error: Tactic failure: reversible in 1st order mode.`
- https://gitlab.com/coq/coq/-/jobs/168148638
  > `make[1]: *** [theories/Reals/Rpower.vo] Segmentation fault (core dumped)`
- https://gitlab.com/coq/coq/-/jobs/167922293
  > `Error: Illegal application: 
The term "@Logic.eq" of type "forall A : Type, A -> A -> Prop"
cannot be applied to the terms
 "Z" : "Set"
 "z" : "Z"
 "[|x|]" : "Z"
The 1st term has type "Set" which should be coercible to 
"Type".`

Let's collect failures here as they tend to be diverse, and let's collect also the test-suite:async failures here.

**Comment by SkySkimmer** [comment]
People talk about our CI on twitter?

**Comment by ejgallego** [comment]
I mean Gitter.

**Comment by ejgallego** [comment]
Failures are happening at an alarming rate (cc: @maximedenes @gares ) thus I'd dare to say that the status of this job in the CI may be demoted to `allow_failure` if things continues like that.

**Comment by ejgallego** [comment]
Failures continue and indeed there seem to be some state handling problems in the STM, for example it is not clear to me if declarations that may involved hooks are correctly determined to be ```Sync`.

**Comment by ejgallego** [comment]
Plenty of failures https://gitlab.com/coq/coq/-/jobs/179101771 https://gitlab.com/coq/coq/-/jobs/179093502 https://gitlab.com/coq/coq/-/jobs/179053653 https://gitlab.com/coq/coq/-/jobs/179043712 https://gitlab.com/coq/coq/-/jobs/178917545

this is just on the first page of gitlab as of now; this misuse of resources is jeopardizing the whole CI infrastructure so I am much afraid that we cannot continue like this for long.

**Comment by maximedenes** [comment]
@gares Any idea how we can debug this?

**Comment by ppedrot** [comment]
The universe errors are not surprising, I believe this is just the non-commutativity of the inference algorithm at work. I wouldn't have thought that this would happen in practice though. Some test failures are probably an instance of this as well. The SEGFAULT is much more concerning in my opinion.

**Comment by gares** [comment]
No, I've no idea. But some of the links are about the `-quick` pass, that has nothing to do with the one doing `coqc -async-proofs on`. What are we talking about here?

The segfault: could it be a stack overflow?

**Comment by ejgallego** [comment]
This is still going on, example

https://gitlab.com/coq/coq/-/jobs/552074644

together with #9637 this makes me wonder about whether it is worth the effort to keep these jobs in the CI, and for example to spend considerable manpower in supporting them in Dune.

It's been 1 year and a half and we have not managed to make these jobs to work reliably in our own CI; also most users report the modes are really not usable due to space constraints.

**Comment by gares** [comment]
Async has nothing to do with vio, that has space overhead.
Async tests what happens when a user open a file with CoqIDE or VScoq and jumps to the end.
